ts/main.ts

```diff
@@ -1 +1,36 @@
 /* global data */
+interface FormElements extends HTMLFormControlsCollection {
+  title: HTMLInputElement;
+  url: HTMLInputElement;
+  notes: HTMLTextAreaElement;
+}
+
+const $urlLinkInput = document.querySelector('#photo-url');
+const $entryForm = document.querySelector('#entry-form') as HTMLFormElement;
+const $previewPhoto = document.querySelector('.preview');
+if (!$urlLinkInput || !$entryForm)
+  throw new Error('$urlLinkInput or $entryForm query failed');
+
+$urlLinkInput.addEventListener('input', (event: Event) => {
+  $urlLinkInput.addEventListener('blur', () => {
+    const eventTarget = event.target as HTMLInputElement;
+    if (eventTarget.value.match(/\.(jpeg|jpg|gif|png)$/)) {
+      $previewPhoto?.setAttribute('src', eventTarget.value);
+    }
+  });
+});
```

Comment:
You don't want to add another event listener inside of here. This is going to cause you to add a `blur` event listener every single time the user types on their keyboard, so you could be creating hundreds or thousands of events.

I see what you are trying to do here with the `blur` event, but I would just keep the `input` event for the functionality.